File: hospital-data-manager/src/main/java/com/example/hospitaldatamanager/components/members/model/Member.java
package com.example.hospitaldatamanager.components.members.model;

import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;


@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "members")
@Setter
public class Member {
    @Id
    private String id;
    private String firstName;
    private String lastName;
    private Address address;
    private Date birthDate;
    private String phone;
    private String cellPhone;
    private Date positiveResultDate;
    private Date recoveryDate;
    private List<Vaccination> vaccinations;
}

File: hospital-dashboard/src/model/Member.ts
export enum Genre {
  ROCK,
  POP,
  RAP,
  CLASSICAL,
}

interface Address{
  city: string;
  street: string;
  numHouse:number;
}
interface Vaccination{
  vacDate:Date ;
 manufacturer:string;
}

export interface Member {
  id: string;
  firstName:string;
  lastName:string;
  address:Address;
  birthDate:Date;
  phone:string;
  cellPhone:string;
  positiveResultDate:Date;
  recoveryDate:Date;
  vaccinations:Vaccination[];
}
